app/src/test/java/gateway/ITShareFile.java

```diff
@@ -0,0 +1,96 @@
+package gateway;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import gateway.config.Config;
+import gateway.controller.CtrlAccountRegister;
+import gateway.controller.CtrlFileCheck;
+import gateway.controller.CtrlShareFile;
+import gateway.services.ServiceAuth;
+import gateway.services.ServiceMetadata;
+import gateway.services.ServiceMetadata.ResSaveFile;
+import gateway.soap.request.Credentials;
+import gateway.soap.request.ReqFile;
+import gateway.soap.request.ReqShareFile;
+import gateway.soap.response.ResSession;
+import gateway.soap.response.ResStatus;
+import java.util.Random;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ITShareFile
+{
+
+	@BeforeEach void setup () { Config.initializeFromEnv (); }
+
+	@Test void ShareFile ()
+	{
+
+		String username1 = UUID.randomUUID ().toString ();
+		String username2 = UUID.randomUUID ().toString ();
+
+		// register
+		String tokenUser1 = registerAndLoginUserSuccess (username1);
+		String tokenUser2 = registerAndLoginUserSuccess (username2);
+
+		// create a file and aux file
+		ResSaveFile resSaveFile1 = createFile (tokenUser1);
+		ResSaveFile resSaveFile2 = createFile (tokenUser2);
+
+		// check files
+		checkFile (resSaveFile1, tokenUser1);
+		checkFile (resSaveFile2, tokenUser2);
```

Comment:
This is doing nothing. 

`CtrlFileCheck.file_check` returns a boolean `ready` as true or false to indicate that a file is fully uploaded or not. (That means that a file has finished being written by a worker in a storage volume).

I guess what you're trying to do it's to <code> <i>mark</i> </code> the file as ready. However this doesn't seem to be necessary when sharing a file since all your tests pass.

Reply:
Yeah, you are right, I was trying to do that! 

I would like to know the opinion of @PedroChaparro about to considering mark as ready or not the file in this case.

Reply:
Hi, it's not necessary to mark the file as ready before sharing it. This isn't a problem because non-ready files aren't sent to users, so, basically, users can't share a non-ready file from the UI because it won't be even shown.

Reply:
Thanks to both of you @PedroChaparro and @Woynert!